test/integration/segmentation_camera_plugin.cc

```diff
@@ -0,0 +1,305 @@
+}
+
+/// \brief Build the scene with 3 boxes besides each other
+/// the 2 aside boxes have the same label & the middle is different
+void BuildScene(rendering::ScenePtr scene)
+{
+  math::Vector3d leftPosition(3, -1.5, 0);
+  math::Vector3d rightPosition(3, 1.5, 0);
+  math::Vector3d middlePosition(3, 0, 0);
+
+  rendering::VisualPtr root = scene->RootVisual();
+
+  // create box visual
+  rendering::VisualPtr box = scene->CreateVisual();
+  box->AddGeometry(scene->CreateBox());
+  box->SetOrigin(0.0, 0.0, 0.0);
+  box->SetLocalPosition(leftPosition);
+  box->SetLocalRotation(0, 0, 0);
+  box->SetUserData("label", 1);
+  root->AddChild(box);
+
+  // create box visual of same label
+  rendering::VisualPtr box1 = scene->CreateVisual();
+  box1->AddGeometry(scene->CreateBox());
+  box1->SetOrigin(0.0, 0.0, 0.0);
+  box1->SetLocalPosition(rightPosition);
+  box1->SetLocalRotation(0, 0, 0);
+  box1->SetUserData("label", 1);
+  root->AddChild(box1);
+
+  // create box visual of different label
+  ignition::rendering::VisualPtr box2 = scene->CreateVisual();
+  box2->AddGeometry(scene->CreateBox());
+  box2->SetOrigin(0.0, 0.0, 0.0);
+  box2->SetLocalPosition(middlePosition);
+  box2->SetLocalRotation(0, 0, 0);
+  box2->SetUserData("label", 2);
+  root->AddChild(box2);
+}
+
+/////////////////////////////////////////////////
+void SegmentationCameraSensorTest::ImagesWithBuiltinSDF(
+  const std::string &_renderEngine)
+{
+  std::string path = ignition::common::joinPaths(PROJECT_SOURCE_PATH, "test",
+      "integration", "segmentation_camera_sensor_builtin.sdf");
+  sdf::SDFPtr doc(new sdf::SDF());
+  sdf::init(doc);
+  ASSERT_TRUE(sdf::readFile(path, doc));
+  ASSERT_NE(nullptr, doc->Root());
+  ASSERT_TRUE(doc->Root()->HasElement("model"));
+  auto modelPtr = doc->Root()->GetElement("model");
+  ASSERT_TRUE(modelPtr->HasElement("link"));
+  auto linkPtr = modelPtr->GetElement("link");
+  ASSERT_TRUE(linkPtr->HasElement("sensor"));
+  auto sensorPtr = linkPtr->GetElement("sensor");
+  ASSERT_TRUE(sensorPtr->HasElement("camera"));
+  auto cameraPtr = sensorPtr->GetElement("camera");
+  ASSERT_TRUE(cameraPtr->HasElement("image"));
+  auto imagePtr = cameraPtr->GetElement("image");
+
+  int width = imagePtr->Get<int>("width");
+  int height = imagePtr->Get<int>("height");
+
+  // Setup ign-rendering with an empty scene
+  auto *engine = ignition::rendering::engine(_renderEngine);
+  if (!engine)
+  {
+    igndbg << "Engine '" << _renderEngine
+              << "' is not supported" << std::endl;
+    return;
+  }
+
+  ignition::rendering::ScenePtr scene = engine->CreateScene("scene");
+  BuildScene(scene);
+
+  ignition::sensors::Manager mgr;
+
+  sdf::Sensor sdfSensor;
+  sdfSensor.Load(sensorPtr);
+
+  std::string type = sdfSensor.TypeStr();
+  EXPECT_EQ(type, "segmentation_camera");
+
+  ignition::sensors::SegmentationCameraSensor *sensor =
+    mgr.CreateSensor<ignition::sensors::SegmentationCameraSensor>(sdfSensor);
+
+  EXPECT_NE(sensor, nullptr);
```

Comment:
```suggestion
  ASSERT_NE(sensor, nullptr);
```
We want to use `ASSERT` here instead of `EXPECT` because if the sensor is `nullptr`, we should stop the test here instead of continuing and de-referencing a `nullptr`.